### PR TITLE
EVA-3263 - Shallow validation

### PR DESCRIPTION
**Pull request by tcezard**

### CODE REVIEW

File: eva_sub_cli/validators/docker_validator.py [hunk 1]
```diff
@@ -38,21 +38,23 @@ def _validation_file_path_for(file_path):
...
-                f"{self.docker_path} exec {self.container_name} nextflow run eva_sub_cli/nextflow/validation.nf "
-                f"--base_dir {container_validation_dir} "
-                f"--vcf_files_mapping {self.mapping_file} "
-                f"--metadata_xlsx {self.metadata_xlsx} "
+            docker_cmd = ''.join([
+                f"{self.docker_path} exec {self.container_name} nextflow run eva_sub_cli/nextflow/validation.nf ",
+                f"--base_dir {container_validation_dir} ",
+                f"--vcf_files_mapping {self.mapping_file} ",
+                f"--metadata_xlsx {self.metadata_xlsx} ",
+                f"--shallow_validation true " if self.shallow_validation else "",
```
Comment (apriltuesday) [inline eva_sub_cli/validators/docker_validator.py:46]: Feels like we could reduce some code repetition by doing this for the xlsx/json check as well, might be messier though - up to you.

File: eva_sub_cli/validators/validator.py [hunk 2]
```diff
@@ -157,105 +153,30 @@ def update_config_with_validation_result(self):
...
+        return all((
+            self.results.get('vcf_check', {}).get('critical_count', 1) == 0,
+            self.results.get('assembly_check', {}).get('nb_mismatch', 1) == 0,
+            self.results.get('assembly_check', {}).get('nb_error', 1) == 0,
+            all((
+                fa_file_check.get('all_insdc', False) is True
+                for fa_file, fa_file_check in self.results.get('fasta_check', {}).items()
+            )),
+            self.results.get('sample_check', {}).get('overall_differences', True) is False,
+            len(self.results.get('metadata_check', {}).get('spreadsheet_errors', [])) == 0,
```
Comment (apriltuesday) [inline eva_sub_cli/validators/validator.py:169]: Should this also check json errors?

**Review by nitin-ebi — APPROVED** [review]

File: eva_sub_cli/jinja_templates/html_report.html [hunk 3]
```diff
@@ -46,6 +47,10 @@ <h6>eva-sub-cli v{{cli_version}}</h6>
     </div>
 </header>
 
+<section>
+    {{ shallow_validation_report(validation_results) }}
+</section>
+
```
[the remark this answers is not in the document]
Comment (tcezard) [reply eva_sub_cli/jinja_templates/html_report.html:53]: Or I move the section to the function which will differ from how the rest of the file is organised but I feel is slightly better.

File: eva_sub_cli/jinja_templates/shallow_validation.html [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    {% if results.get('required') %}
+        <div class="report-section fail collapsible"> <span class="expand_icon">&#9654;</span>
+            &#10060; <b>You requested to run the shallow validation, please run full validation before submitting the data</b>
+        </div>
+        <div class="no-show">
+        <table>
+            <tr>
+                <th><strong>VCF File</strong></th>
+                <th><strong>Records validated in VCF</strong></th>
+                <th><strong>Records validated in Fasta</strong></th>
```
[the remark this answers is not in the document]
Comment (tcezard) [reply eva_sub_cli/jinja_templates/shallow_validation.html:14]: Changed it.

File: eva_sub_cli/validators/docker_validator.py (hunk 1 above)
Comment (tcezard) [reply eva_sub_cli/validators/docker_validator.py:46]: I changed it and tried to keep it readable

File: eva_sub_cli/validators/validator.py (hunk 2 above)
Comment (tcezard) [reply eva_sub_cli/validators/validator.py:169]: Yes it should have.

**Review by apriltuesday — APPROVED** [review]

**Pull request merged**